src/terms/Hamiltonian.jl

```diff
@@ -2,121 +2,149 @@
 # corresponding to each term
 # This is the "high-level" interface, provided for convenience
 
-struct HamiltonianBlock
+abstract type HamiltonianBlock end
+
+# Generic HamiltonianBlock
+struct GenericHamiltonianBlock <: HamiltonianBlock
     basis::PlaneWaveBasis
     kpoint::Kpoint
 
     # The operators are vectors of RealFourierOperator,
     # not typed because of type invariance issues.
     operators::Vector  # the original list of RealFourierOperator
                        # (as many as there are terms), kept for easier exploration
-    optimized_operators::Vector  # the optimized list of RealFourierOperator, to be used for applying
+    optimized_operators::Vector  # Optimized list of RealFourierOperator, for application
```

Comment:
just don't do the optimization in the generic hamiltonian code? Makes it even clearer